src/fw-logs/fw-string-formatter.cpp

```diff
@@ -135,16 +127,57 @@ namespace librealsense
                                 s << "Protocol Error recognized!\nImproper log message received: " << source_temp
                                     << ", invalid parameter: " << exp_replace_it->second << ".\n The range of supported values is \n";
                                 for_each(vec.begin(), vec.end(), [&s](kvp& entry) { s << entry.first << ":" << entry.second << " ,"; });
-                                std::cout << s.str().c_str() << std::endl;;
+                                LOG_WARNING( s.str() );
                             }
                             source_temp = destTemp;
                         }
                     }
                 }
             }
 
-            *dest = source_temp;
+            return source_temp;
+        }
+
+        std::string fw_string_formatter::convert_param_to_string( const param_info & info, const uint8_t * param_start ) const
+        {
+            switch( info.type )
+            {
+            case param_type::STRING:
+                return std::string( reinterpret_cast< const char * >( param_start ), info.size );
+            case param_type::UINT8:
+                return rsutils::string::from() << *reinterpret_cast< const uint8_t * >( param_start );
+            case param_type::SINT8:
+                return rsutils::string::from() << *reinterpret_cast< const int8_t * >( param_start );
+            case param_type::UINT16:
+                return rsutils::string::from() << *reinterpret_cast< const uint16_t * >( param_start );
+            case param_type::SINT16:
+                return rsutils::string::from() << *reinterpret_cast< const int16_t * >( param_start );
+            case param_type::SINT32:
+                return rsutils::string::from() << *reinterpret_cast< const int32_t * >( param_start );
+            case param_type::UINT32:
+                return rsutils::string::from() << *reinterpret_cast< const uint32_t * >( param_start );
+            case param_type::SINT64:
+                return rsutils::string::from() << *reinterpret_cast< const int64_t * >( param_start );
+            case param_type::UINT64:
+                return rsutils::string::from() << *reinterpret_cast< const uint64_t * >( param_start );
+            case param_type::FLOAT:
+                return rsutils::string::from() << *reinterpret_cast< const float * >( param_start );
+            case param_type::DOUBLE:
+                return rsutils::string::from() << *reinterpret_cast< const double * >( param_start );
+            default:
+                throw librealsense::invalid_value_exception( rsutils::string::from()
+                                                             << "Unsupported parameter type "
+                                                             << static_cast< uint8_t >( info.type ) );
+            }
+        }
+
+        bool fw_string_formatter::is_integral( const param_info & info ) const
```

Comment:
Do you meant "is_integer"?

Reply:
"integral" means "of integer type".
`is_ineger` might be confusing in case of unsigned.